homeassistant/components/ultrasync/sensor.py

```diff
@@ -0,0 +1,176 @@
+"""Monitor the Interlogix/Hills ComNav UltraSync Hub."""
+
+import logging
+from typing import Callable, List
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import UltraSyncEntity
+from .const import (
+    DATA_COORDINATOR,
+    DATA_UNDO_UPDATE_LISTENER,
+    DOMAIN,
+    SENSOR_UPDATE_LISTENER,
+    SENSORS,
+)
+from .coordinator import UltraSyncDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], bool], None],
+) -> None:
+    """Set up UltraSync sensor based on a config entry."""
+    coordinator: UltraSyncDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    # At least one sensor must be pre-created or Home Assistant will not
+    # call any updates
+    hass.data[DOMAIN][entry.entry_id][SENSORS]["area01_state"] = UltraSyncSensor(
```

Comment:
Why is this 'dummy' sensor created? In my opinion, it would make more sense to pull all available areas first in __init__.py and create the sensors from there. Eventually they could all pull their latest data from the DataUpdateCoordinator.

Reply:
This was required through advice provided to me by @raman325 who was a fantastic resource on the Home Assistant Discord chat at the time.  Without per-creating a dummy sensor to work with nothing ever gets detected after. It's like a ticket into the event loop. If i don't create one at the initialization, future changes never get triggered.

If i do this, everything works great afterwards.